main.cpp

```diff
@@ -250,7 +487,10 @@ struct ping_queue_t
 
 		*out = m_queue.front();
 		m_queue.pop_front();
-		m_ips.erase(out->ep);
+		if (out->ep.protocol() == udp::v4())
```

Comment:
I would have expected that you'd basically turn this whole class into a template and have two ping_queues. it seems this would introduce a lot of if-statements